packages/core/src/components/button/_button.scss

```diff
@@ -226,14 +226,14 @@ Styleguide button
 
 a.#{$ns}-button {
   text-align: center;
-  text-decoration: none;
   transition: none;
 
   &,
   &:hover,
   &:active {
     // override global 'a' styles
     color: $pt-text-color;
+    text-decoration: none;
```

Comment:
I can imagine this causing a styling regression in some rare case where a Blueprint user is trying to create a kind of "button link" component which _does_ show a text underline (cue [relevant xkcd](https://xkcd.com/1172/)), but I think this change is the safest way to fix the problem at hand and should generally be OK. We are planning to add a "button link" component to the design system soon which should address that use case anyway.